spec/javascripts/LoginPage/Login3scaleForm.spec.jsx

```diff
@@ -27,18 +27,62 @@ it('should render HiddenInputs component', () => {
   expect(wrapper.find(HiddenInputs).exists()).toEqual(true)
 })
 
-it('should set username state', () => {
-  const wrapper = mount(<Login3scaleForm {...props}/>)
-  jest.spyOn(wrapper.instance(), 'handleTextInputUsername')
-  wrapper.instance().handleTextInputUsername('foo')
-  expect(wrapper.instance().handleTextInputUsername).toHaveBeenCalled()
-  expect(wrapper.state().username).toEqual('foo')
+describe('username', () => {
+  it('should set username and validation state to true', () => {
+    const event = {
+      currentTarget: {
+        name: 'username',
+        value: 'Bob',
+        type: 'text'
+      }
+    }
+    const wrapper = mount(<Login3scaleForm {...props}/>)
+    wrapper.find('input#session_username').props().onChange(event)
+    expect(wrapper.state('username')).toEqual('Bob')
+    expect(wrapper.state('validation').username).toEqual(true)
+  })
+  it('should set validation username state to false when input value is missing', () => {
+    const event = {
+      currentTarget: {
+        name: 'username',
+        value: '',
+        type: 'text'
+      }
+    }
+    const wrapper = mount(<Login3scaleForm {...props}/>)
+    wrapper.find('input#session_username').props().onChange(event)
+    expect(wrapper.state().username).toEqual('')
+    expect(wrapper.state('validation').username).toEqual(false)
```

Comment:
I know it's not a big deal but let's try to be consistent, use same kind of accessor:
* `state().username`, `state().validation.username`
* `state('username')`, `state('validation').username`